### PR TITLE
Lets people break apart guns for gun parts

### PR DESCRIPTION
**Pull request by MLGTASTICa**
## About The Pull Request
Adds the ability to break apart guns for their parts using a wrench
## Why It's Good For The Game
Cause a staff dev asked for it , also adds another way to destroy "oddity" guns like the skydriver and complete the objective

## Changelog
:cl:
add: Added the ability to break apart guns for their parts and salvage.
/:cl:

### CODE REVIEW

**Comment by Wouju** [comment]
looks good
suggesting that deconstructing the sky driver gives something more, though that can be done later

File: code/modules/projectiles/guns/projectile/boltgun.dm [hunk 1]
```diff
@@ -103,6 +103,7 @@
 	matter = list(MATERIAL_STEEL = 20, MATERIAL_WOOD = 10)
 	wielded_item_state = "_doble_wood"
 	spawn_blacklisted = FALSE
+	gun_parts = list(/obj/item/part/gun = 3 ,/obj/item/stack/material/steel = 16)
```
Comment (Firefox13) [inline code/modules/projectiles/guns/projectile/boltgun.dm:106]: Novakovic doesn't even uses plasteel yet dismantling the gun parts will give us some, let's stick to thermodynamics laws and won't create matter out of nothing. It's so irredemeably shitty that it shouldn't yield any. Same for kardashev-mosin.

File: code/modules/projectiles/guns/projectile/automatic/lmg.dm [hunk 2]
```diff
@@ -25,6 +25,7 @@
 	one_hand_penalty = 30 //you're not Stallone. LMG level.
 	spawn_blacklisted = TRUE
 	rarity_value = 80
+	gun_parts = list(/obj/item/part/gun = 1 ,/obj/item/stack/material/steel = 4)
```
Comment (Firefox13) [inline code/modules/projectiles/guns/projectile/automatic/lmg.dm:28]: funnily enough steel isn't even used in building LMGs and dismantling them shouldn't yield any

File: code/modules/projectiles/gun.dm [hunk 3]
```diff
@@ -80,6 +81,23 @@
...
 	var/inversed_carry = FALSE
 
+/obj/item/weapon/gun/attackby(obj/item/I, mob/living/user, params)
+	if(!istool(I))
+		return FALSE
+	if(!gun_parts)
+		to_chat(user, SPAN_NOTICE("You can't dismantle [src] it has no gun  parts!"))
+		return FALSE
+	if(I.get_tool_quality(QUALITY_BOLT_TURNING))
+		user.visible_message(SPAN_NOTICE("[user] begins breaking apart [src]"), SPAN_NOTICE("You begin breaking apart [src] for gun parts"))
```
Comment (Firefox13) [inline code/modules/projectiles/gun.dm:91]:
```suggestion
		user.visible_message(SPAN_NOTICE("[user] begins breaking apart [src]."), SPAN_NOTICE("You begin breaking apart [src] for gun parts."))
```

File: code/modules/projectiles/gun.dm [hunk 4]
```diff
@@ -80,6 +81,23 @@
...
+	if(I.use_tool(user, src, WORKTIME_SLOW, QUALITY_BOLT_TURNING, FAILCHANCE_EASY, required_stat = STAT_MEC))
+		user.visible_message(SPAN_NOTICE("[user] breaks [src] apart for gun parts!."), SPAN_NOTICE("You break [src] apart for gun parts."))
```
Comment (Firefox13) [inline code/modules/projectiles/gun.dm:93]:
```suggestion
		user.visible_message(SPAN_NOTICE("[user] breaks [src] apart for gun parts!"), SPAN_NOTICE("You break [src] apart for gun parts."))
```

File: code/modules/projectiles/gun.dm [hunk 5]
```diff
@@ -80,6 +81,23 @@
 	var/noricochet = FALSE // wether or not bullets fired from this gun can ricochet off of walls
 	var/inversed_carry = FALSE
 
+/obj/item/weapon/gun/attackby(obj/item/I, mob/living/user, params)
+	if(!istool(I))
+		return FALSE
+	if(!gun_parts)
+		to_chat(user, SPAN_NOTICE("You can't dismantle [src] it has no gun  parts!"))
```
Comment (Firefox13) [inline code/modules/projectiles/gun.dm:88]: If this message appears it probably means something went horribly wrong and admins should be notified
```suggestion
		to_chat(user, SPAN_NOTICE("You can't dismantle [src] as it has no gun parts! How strange..."))
```

**Review by Firefox13 — CHANGES_REQUESTED** [review]
Was it actually tested? Does any gun yields parts from dismantling? What about extra fancy guns, are you planning on making them yield extra fancy materials? Energy guns? Are all guns don't create materials out of nowhere and respect thermodynamics laws? What happens if you try to dismantle a modded gun with combi-driver(both screwdriving and bolt turning tiers)?

**Comment by Wouju** [comment]
Firefox is a coward who respects the laws of thermodynamics

Cringe!

**Comment by MLGTASTICa** [comment]
> Was it actually tested? Does any gun yields parts from dismantling? What about extra fancy guns, are you planning on making them yield extra fancy materials? Energy guns? Are all guns don't create materials out of nowhere and respect thermodynamics laws? What happens if you try to dismantle a modded gun with combi-driver(both screwdriving and bolt turning tiers)?

Yes it was tested . All subtypes of gun yield parts from dismantling , if you give me ideas for extra fancy materials sure. Energy guns also yield parts . Most guns respect the laws of "thermodynamics". Trying to dismantle or unmod a gun with both starts both dismantling and lets you see the unmod menu.

**Comment by Evankhell561** [comment]
> Most guns respect the laws of "thermodynamics". Trying to dismantle or unmod a gun with both starts both dismantling and lets you see the unmod menu.

If you want to preserve the laws of thermodynamics, you should make the total amount of materials that the gun parts have an equal or lesser amount than the original gun (you can even change the type of material a gun part is made of). 

This is really important for weapons that have been printed on an autolathe.

**Comment by Firefox13** [comment]
> Trying to dismantle or unmod a gun with both starts both dismantling and lets you see the unmod menu.

Well, can we do something about it?

**Comment by MLGTASTICa** [comment]
> > Trying to dismantle or unmod a gun with both starts both dismantling and lets you see the unmod menu.
> 
> Well, can we do something about it?

ill make it require harm intent

**Comment by MLGTASTICa** [comment]
> looks good
> suggesting that deconstructing the sky driver gives something more, though that can be done later

it now gives 3 gauss coil barrels.

File: code/modules/projectiles/guns/projectile/boltgun.dm [hunk 6]
```diff
@@ -103,6 +103,7 @@
 	matter = list(MATERIAL_STEEL = 20, MATERIAL_WOOD = 10)
 	wielded_item_state = "_doble_wood"
 	spawn_blacklisted = FALSE
+	gun_parts = list(/obj/item/stack/material/steel = 16)
```
Comment (Firefox13) [inline code/modules/projectiles/guns/projectile/boltgun.dm:106]: Do the same to Kardashev-Mosin?

Comment (MLGTASTICa) [reply code/modules/projectiles/guns/projectile/boltgun.dm:107]: Done

**Review by Firefox13 — APPROVED** [review]

**Pull request merged**